core/embed/io/display/ltdc_dsi/display_driver.c

```diff
@@ -0,0 +1,404 @@
+
+#include <trezor_bsp.h>
+#include <trezor_model.h>
+#include <trezor_rtl.h>
+
+#include <io/display.h>
+
+#ifdef KERNEL_MODE
+
+#include <sys/irq.h>
+#include <sys/mpu.h>
+#include <sys/systick.h>
+
+#ifdef USE_BACKLIGHT
+#include "../backlight/backlight_pwm.h"
+#endif
+
+#include "display_internal.h"
+
+display_driver_t g_display_driver = {
+    .initialized = false,
+};
+
+static void display_pll_init(void) {
+  RCC_PeriphCLKInitTypeDef PLL3InitPeriph = {0};
+
+  /* Start and configure PLL3 */
+  /* HSE = 16/32MHZ */
+  /* 16/32/(M=8)   = 4MHz input (min) */
+  /* 4*(N=125)  = 500MHz VCO (almost max) */
+  /* 500/(P=8)  = 62.5 for DSI ie exactly the lane byte clock*/
+
+  PLL3InitPeriph.PeriphClockSelection = RCC_PERIPHCLK_DSI | RCC_PERIPHCLK_LTDC;
+  PLL3InitPeriph.DsiClockSelection = RCC_DSICLKSOURCE_PLL3;
+  PLL3InitPeriph.LtdcClockSelection = RCC_LTDCCLKSOURCE_PLL3;
+#if HSE_VALUE == 32000000
+  PLL3InitPeriph.PLL3.PLL3M = 8;
+#elif HSE_VALUE == 16000000
+  PLL3InitPeriph.PLL3.PLL3M = 4;
+#endif
+  PLL3InitPeriph.PLL3.PLL3N = 125;
+  PLL3InitPeriph.PLL3.PLL3P = 8;
+  PLL3InitPeriph.PLL3.PLL3Q = 8;
+  PLL3InitPeriph.PLL3.PLL3R = 24;
+  PLL3InitPeriph.PLL3.PLL3FRACN = 0;
+  PLL3InitPeriph.PLL3.PLL3RGE = RCC_PLLVCIRANGE_0;
+  PLL3InitPeriph.PLL3.PLL3ClockOut = RCC_PLL3_DIVR | RCC_PLL3_DIVP;
+  PLL3InitPeriph.PLL3.PLL3Source = RCC_PLLSOURCE_HSE;
+  (void)HAL_RCCEx_PeriphCLKConfig(&PLL3InitPeriph);
+}
+
+static void display_dsi_init(void) {
```

Comment:
checking the values here: https://github.com/trezor/trezor-firmware/commit/f7117a70f8209681f8e4d42ba897b206ae92fb59

as `display_init` does not return anything i just reset the peripheral and set display uninitialized in case of any failure.